lib/miou_sync.ml

```diff
@@ -52,8 +52,8 @@ end = struct
     | Awaiting (fn, x, y) as seen ->
         if Atomic.compare_and_set t seen Signaled then fn t x y else signal t
 
-  let is_signaled t = Atomic.get t == Signaled
-  let is_initial t = Atomic.get t == Initial
+  let is_signaled t = Atomic.get t = Signaled
+  let is_initial t = Atomic.get t = Initial
```

Comment:
Note that `==` and `=` will do the exact same thing in this case.

Reply:
Yes, but I prefer to be conservative on this space. I mean, it's hard to remember all reasons that it's currently safe to user `==`  instead of `=` :confused:.

Reply:
Note that `Atomic.compare_and_set` uses physical equality.  I would generally recommend thinking in terms of physical equality with atomics unless there is very specific reason to prefer other forms of equality.

Reply:
This is indeed the model I could have had in the first place, but [this issue](https://github.com/ocaml/ocaml/issues/12713) still makes me break out in a cold sweat over the use of `==.` Once again, the point is to facilitate rereading without reconsidering why it's okay to use `==` 10 years after you've done it!